packages/block-library/src/query/index.php

```diff
@@ -137,10 +138,16 @@ function block_core_query_disable_enhanced_pagination( $parsed_block ) {
 	static $dirty_enhanced_queries = array();
 	static $render_query_callback  = null;
 
-	$is_interactive = isset( $parsed_block['attrs']['enhancedPagination'] ) && true === $parsed_block['attrs']['enhancedPagination'] && isset( $parsed_block['attrs']['queryId'] );
-	$block_name     = $parsed_block['blockName'];
+	$block_name   = $parsed_block['blockName'];
+	$block_object = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
+
+	$is_interactive_bool   = isset( $block_object->supports['interactivity'] ) && true === $block_object->supports['interactivity'];
+	$is_interactive_object = isset( $block_object->supports['interactivity']['interactive'] ) && true === $block_object->supports['interactivity']['interactive'];
+	$client_navigation     = isset( $block_object->supports['interactivity']['clientNavigation'] ) && true === $block_object->supports['interactivity']['clientNavigation'];
```

Comment:
This is PHP 5.6 compatible.

This change would not work:
```
	$is_interactive_bool   = (bool) ( $block_object->supports['interactivity'] ?? false );
	$is_interactive_object = (bool) ( $block_object->supports['interactivity']['interactive'] ?? false );
	$client_navigation     = (bool) ( $block_object->supports['interactivity']['clientNavigation'] ?? false );
```

Reply:
[WordPress does not support PHP 5.6 anymore.](https://make.wordpress.org/core/handbook/references/php-compatibility-and-wordpress-versions/)